rustls/src/server/common.rs

```diff
@@ -30,16 +30,16 @@ impl HandshakeDetails {
 }
 
 pub struct ServerKXDetails {
-    pub kx: Option<kx::KeyExchange>,
+    pub kx: kx::KeyExchange,
 }
 
 impl ServerKXDetails {
     pub fn new(kx: kx::KeyExchange) -> ServerKXDetails {
-        ServerKXDetails { kx: Some(kx) }
+        ServerKXDetails { kx }
     }
 
-    pub fn take_kx(&mut self) -> kx::KeyExchange {
-        self.kx.take().unwrap()
+    pub fn take_kx(self) -> kx::KeyExchange {
```

Comment:
Do we still need this function? Field `kx` is already pub so I think the callers of this function can just move it out of the structure. One of the goals of my idea is to remove the need for these `take_*` functions.

I.e., instead of `let kx = self.server_kx.take_kx()`, we can just write `let kx = self.server_kx.kx` or even maybe remove the local `kx` variable and use `self.server_kx` directly where it is used.

Reply:
The second idea sounds good. I had kept this in spirit of more minimal changes, but I've removed the method in favor of a direct field access now.